### DISCUSSION TITLE
Debian support dropped for i386?

### DISCUSSION THREAD

**Discussion by willew** [hertzg/rtl_433_docker#87] [General]
My rtl_433 docker stopped working app 1 month ago and I believe (not sure) it was related to the image update done with docker compose. 

First I believed that it is a hardware issue, since the non docker version started to work after re-plugging the tuner. But after some additional testing, I am quite sure that the non-docker version (app 1 year old version) works ok and the docker version will not work and will mess up the tuner requiring re-plugging.

I have Ubuntu 20.04 running in Intel NUC i3. When I checked the documentation, Intel was not supported for Debian.

Have the support for Intel/Debian dropped  in one of the versions? I do not know how to check versions in Github. If yes, can someone help me by telling how to use the older version supporting still Intel?

**Comment by hertzg** [comment]
I'm not sure how to help you as there can be multitude of reasons why it could brake or not...

**Comment by willew** [comment]
If I have Intel NUC with Intel i3 and Ubuntu 20, should the docker version work? x84 with Debian is listed as unsupported. Has this been always like this?

**Reply by hertzg** [reply]
All Intel i3 processors are x86-64 systems, why do you think it would not work? Where did you see the that "support list"?

**Comment by willew** [comment]
In the end of the documentation there is a list of supporter platforms.

Architecture | Alpine | Debian
linux/386 | ✔️ | ➖

To me this says that intel is supporter on Alpine, but not Debian

**Reply by hertzg** [reply]
https://en.wikipedia.org/wiki/I386 is but https://en.wikipedia.org/wiki/IA-32 is not synonymous to Intel which is sometimes called `i386`, it's referencing the 32 bit architecture (and more but details are irrelevant for this discussion). Intel also has processors that are `amd64` based which is also called `x86_64` sometimes.

Now regarding support of `i386` on debian, last I remember the build of `rtl433` was failing therefore I had to disable it. If you have an `386` machine and you manage to build the docker image there, feel free to push a PR, I'll be happy to review it and help get merged.

**Comment by willew** [comment]
I thought that 386 refers to any Intel platform and amd64 refers to AMD processors, my fault. My Ubuntu  is the following. The simple question is that is this supported by default and should I start to investigate other reasons why it fails? Currently I am using the non-docker version ran a service as a workaround.

```
$ lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   36 bits physical, 48 bits virtual
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              2
Core(s) per socket:              2
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           58
Model name:                      Intel(R) Core(TM) i3-3217U CPU @ 1.80GHz
```

**Reply by hertzg** [reply]
Anything after core duo (if memory serves me right) is 64 bits therefore x86_64 which is also called amd64. If your platform was not supported you would not be able to pull the docker image. So the problem is not there.

**Comment by hertzg** [comment]
Closing as resolved
